Makefile

```diff
@@ -4,14 +4,81 @@
 DEBUG  ?= 0
 OPENMP ?= 1
 MPI    ?= 0
-CXX     = g++
+CXX     = g++ 
 MKDIR_P = mkdir -p
 RM      = /bin/rm
 
 MAKE_CPPFLAGS = -D_GNU_SOURCE -Iinclude -I${GTEST_ROOT}/include
 MAKE_CXXFLAGS = -Wall -std=c++11
 MAKE_LDFLAGS  = -L. -L${GTEST_ROOT} -lgtest
 
+
+
+ifeq ($(PE_ENV), CRAY)
+CXX = CC
+MAKE_CPPFLAGS = -Iinclude -I${GTEST_ROOT}/include
+MAKE_CXXFLAGS = -hstd=c++11 -Iinclude -I/include
+MAKE_LDFLAGS  = -L. -L${GTEST_ROOT} -lgtest -dynamic
+
+ifeq ($(DEBUG), 1)
+       MAKE_CPPFLAGS += -D_DEBUG_
+       MAKE_CXXFLAGS += -G0
+endif
+
+ifeq ($(MPI), 1)
+       MAKE_CPPFLAGS += -DUSE_MPI
+endif
+
+endif
+
+
+ifeq ($(PE_ENV), INTEL)
+CXX = CC
+MAKE_CPPFLAGS = -Iinclude -I${GTEST_ROOT}/include
+MAKE_CXXFLAGS = -std=c++11 -Iinclude -I/include
+MAKE_LDFLAGS  = -L. -L${GTEST_ROOT} -lgtest -dynamic
+
+ifeq ($(DEBUG), 1)
+       MAKE_CPPFLAGS += -D_DEBUG_
+       MAKE_CXXFLAGS += -G0
+endif
+
+ifeq ($(OPENMP), 1)
+	MAKE_CXXFLAGS += -qopenmp
+	MAKE_LDFLAGS  += -qopenmp
+endif
+
+ifeq ($(MPI), 1)
+       MAKE_CPPFLAGS += -DUSE_MPI
+endif
+
+endif
+
+
+ifeq ($(PE_ENV), PGI)
+CXX = CC
+MAKE_CPPFLAGS = -Iinclude -I${GTEST_ROOT}/include
+MAKE_CXXFLAGS = -std=c++11 -Iinclude -I/include
+MAKE_LDFLAGS  = -L. -dynamic
+
+ifeq ($(DEBUG), 1)
+       MAKE_CPPFLAGS += -D_DEBUG_
+       MAKE_CXXFLAGS += -G0
+endif
+
+ifeq ($(OPENMP), 1)
+	MAKE_CXXFLAGS += -mp=nonuma
+	MAKE_LDFLAGS  += -mp=nonuma
+endif
+
+ifeq ($(MPI), 1)
+       MAKE_CPPFLAGS += -DUSE_MPI
+endif
```

Comment:
This `ifeq` should not be duplicated for each compiler. It is just setting the preprocessor flags.